TLM/TLM/State/Flags.cs

```diff
@@ -1072,28 +1072,32 @@ public class Flags {
         }
 
         internal static void OnLevelUnloading() {
-            for (uint i = 0; i < laneConnections.Length; ++i) {
-                laneConnections[i] = null;
-            }
+            try {
+                for (uint i = 0; i < laneConnections.Length; ++i) {
+                    laneConnections[i] = null;
+                }
 
-            for (uint i = 0; i < laneSpeedLimitArray.Length; ++i) {
-                laneSpeedLimitArray[i] = null;
-            }
+                for (uint i = 0; i < laneSpeedLimitArray.Length; ++i) {
+                    laneSpeedLimitArray[i] = null;
+                }
 
-            lock(laneSpeedLimitLock) {
-                laneSpeedLimit.Clear();
-            }
+                lock (laneSpeedLimitLock) {
+                    laneSpeedLimit.Clear();
+                }
 
-            for (uint i = 0; i < laneAllowedVehicleTypesArray.Length; ++i) {
-                laneAllowedVehicleTypesArray[i] = null;
-            }
+                for (uint i = 0; i < laneAllowedVehicleTypesArray.Length; ++i) {
+                    laneAllowedVehicleTypesArray[i] = null;
+                }
 
-            for (uint i = 0; i < laneArrowFlags.Length; ++i) {
-                laneArrowFlags[i] = null;
-            }
+                for (uint i = 0; i < laneArrowFlags.Length; ++i) {
+                    laneArrowFlags[i] = null;
+                }
 
-            for (uint i = 0; i < highwayLaneArrowFlags.Length; ++i) {
-                highwayLaneArrowFlags[i] = null;
+                for (uint i = 0; i < highwayLaneArrowFlags.Length; ++i) {
+                    highwayLaneArrowFlags[i] = null;
+                }
+            } catch (Exception ex) {
+                ex.LogException(true);
```

Comment:
No point in logging the exception here if you can just let it bubble up. The caller also has a try catch. And the caller of that caller also does. The whole point of exceptions is to be able to write the code as if there is no error handling because exceptions should be ... you know .. exceptional.
You only handle them where appropriate. Handling means actually "fix" the issue locally. If you want to catch it so that the application does not crash with an unhandled exception, let it bubble up and "handle / log" it at the top most level only.

Reply:
I was trying to address this comment https://github.com/CitiesSkylinesMods/TMPE/pull/1096#discussion_r622511588

It does make sense. after going to the main menu TMPE will be destroyed so even if managers are not destroyed properly, they are gone (actually they are stored in a singleton instance which is alive forever but we can fix that in the future).

- soft failure: Generally in TMPE we have a strategy to try to continue after failure in the hope that it will work anyway (that is what the existing code does). 
- hard failure: An alternative strategy would be to stop the application in case of failure because the application may be in an unstable state after assertion. if we want to do this we need to write code for an exception dialog that exits the application once the user closes it (or something like that).

I don't think this PR is the appropriate place to make such decisions. If you are interested please create an issue with the label `discussion` to discuss pros and cons of both strategies.

Reply:
plz move try catch to the caller then if possible.